File: archive/kubejs/startup_scripts/multiblock_casings_materials.js

const GTMaterials = Java.loadClass('com.gregtechceu.gtceu.common.data.GTMaterials')
const GTMaterialFlags = Java.loadClass('com.gregtechceu.gtceu.api.data.chemical.material.info.MaterialFlags')

GTCEuStartupEvents.materialModification(event => {
  GTMaterials.Niobium.addFlags(GTMaterialFlags.GENERATE_PLATE)
  GTMaterials.Kanthal.addFlags(GTMaterialFlags.GENERATE_FOIL)
})

StartupEvents.registry('item', event => {
  event.create('heatproof_alloy_mesh')
    .displayName('Heatproof Alloy Mesh')
    .maxStackSize(64);

  event.create('fine_stainless_mesh')
    .displayName('Fine Stainless Mesh')
    .maxStackSize(64);

  event.create('ceramic_filter')
    .displayName('Ceramic Filter')
    .maxStackSize(64);

  event.create('carbon_nanotubes')
    .displayName('Carbon Nanotubes')
    .maxStackSize(64);
});
GTCEuStartupEvents.registry('gtceu:material', event => {

  event.create('ceramic_fiber')
    .dust()
    .components('1x alumina', '2x silicon_dioxide')
    .color(0xEDE9E3)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.GENERATE_PLATE, GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('high_temperature_insulation')
    .dust()
    .components('1x ceramic_fiber', '1x silicon_dioxide')
    .color(0xC8C1B8)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('hexafluoropropylene')
    .gas()
    .components('3x carbon', '6x fluorine')
    .color(0xCFEFFF)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('perfluoroalkoxy_polymer_slurry')
    .liquid()
    .components('5x carbon', '12x fluorine', '1x oxygen')
    .color(0xE6F2FF)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('fluoropolymer_structural_film')
    .polymer()
    .components('5x carbon', '12x fluorine', '1x oxygen')
    .color(0xF7FBFF)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('fluoropolymer_residue')
    .liquid()
    .components('3x carbon', '6x fluorine')
    .color(0xB6C6D6)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('phthalic_anhydride')
    .dust()
    .components('8x carbon', '4x hydrogen', '3x oxygen')
    .color(0xF1E6D6)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('trimellitic_anhydride')
    .dust()
    .components('9x carbon', '4x hydrogen', '5x oxygen')
    .color(0xE8D7C6)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('pyromellitic_dianhydride')
    .dust()
    .components('10x carbon', '2x hydrogen', '6x oxygen')
    .color(0xE6D2B8)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('oxydianiline')
    .dust()
    .components('12x carbon', '12x hydrogen', '2x nitrogen', '1x oxygen')
    .color(0xE9DAB8)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('m_phenylenediamine')
    .dust()
    .components('6x carbon', '8x hydrogen', '2x nitrogen')
    .color(0xD8C7A0)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('bisphenol_a_dianhydride')
    .dust()
    .components('31x carbon', '18x hydrogen', '8x oxygen')
    .color(0xD7C1A3)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('polyetherimide')
    .liquid()
    .components('37x carbon', '22x hydrogen', '2x nitrogen', '6x oxygen')
    .color(0xC79B43)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('polyimide_solution')
    .liquid()
    .components('22x carbon', '10x hydrogen', '2x nitrogen', '5x oxygen')
    .color(0xA06A2F)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('microporous_polyimide_membrane')
    .polymer()
    .components('22x carbon', '10x hydrogen', '2x nitrogen', '5x oxygen')
    .color(0xC9D2DA)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('polysiloxane')
    .liquid()
    .components('2x carbon', '6x hydrogen', '1x oxygen', '1x silicon')
    .color(0xDDE7EF)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('iron_pentacarbonyl')
    .liquid()
    .components('1x iron', '5x carbon', '5x oxygen')
    .color(0xC9D4DC);

  event.create('as_grown_multiwalled_carbon_nanotubes')
    .dust()
    .components('1x carbon')
    .color(0x141414)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('purified_multiwalled_carbon_nanotubes')
    .dust()
    .components('1x carbon')
    .color(0x101010)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('m_dinitrobenzene')
    .dust()
    .components('6x carbon', '4x hydrogen', '2x nitrogen', '4x oxygen')
    .color(0xD9C86A)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('p_chloronitrobenzene')
    .dust()
    .components('6x carbon', '4x hydrogen', '1x chlorine', '1x nitrogen', '2x oxygen')
    .color(0xD8C05E)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('p_nitrophenol')
    .dust()
    .components('6x carbon', '5x hydrogen', '1x nitrogen', '3x oxygen')
    .color(0xC9B04F)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);
  
  event.create('dinitrodiphenyl_ether_4_4')
    .dust()
    .components('12x carbon', '8x hydrogen', '2x nitrogen', '5x oxygen')
    .color(0xC9B36B)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('phenolic_resin')
    .polymer()
    .components('7x carbon', '6x hydrogen', '1x oxygen')
    .color(0x6B3C22)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('carbon_matrix_composite')
    .dust()
    .components('1x carbon', '1x graphite')
    .color(0x1C1C1C)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('molten_volcanic_glass')
    .liquid()
    .components('2x silicon_dioxide', '1x alumina')
    .color(0x2A202B)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('obsidian_ceramic_composite')
    .dust()
    .components('2x silicon_dioxide', '1x alumina')
    .color(0x3B303E)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('aluminosilicate_gel')
    .dust()
    .components('1x aluminum', '1x silicon', '4x oxygen', '2x hydrogen')
    .color(0xC8D0C8)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('zeolite_catalyst_matrix')
    .dust()
    .components('1x sodium', '1x aluminum', '1x silicon', '4x oxygen', '2x hydrogen')
    .color(0xB9C7B7)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

  event.create('catalytic_cracking_matrix')
    .dust()
    .components('1x zeolite_catalyst_matrix', '1x vanadium_pentoxide', '1x alumina')
    .color(0x879A86)
    .iconSet(GTMaterialIconSet.DULL)
    .flags(GTMaterialFlags.DISABLE_DECOMPOSITION);

});

File: gregorius-drugworks-workspace/gregorius-drugworks-gtpersistence/src/main/java/com/wurtzitane/gregoriusdrugworkspersistence/recipe/MultiblockCasingsMaterials.java
package com.wurtzitane.gregoriusdrugworkspersistence.recipe;

import gregtech.api.unification.material.Material;
import gregtech.api.unification.material.Materials;

import static gregtech.api.unification.material.info.MaterialFlags.*;
import static gregtech.api.unification.material.info.MaterialIconSet.*;
import static com.wurtzitane.gregoriusdrugworkspersistence.recipe.GregoriusDrugworksMaterials.resLoc;

public final class MultiblockCasingsMaterials {
    private MultiblockCasingsMaterials() {}

    public static void init() {
        GregoriusDrugworksMaterials.CeramicFiber = new Material.Builder(32000, resLoc("ceramic_fiber"))
                .dust()
                .color(0xEDE9E3)
                .iconSet(DULL)
                .flags(GENERATE_PLATE, DISABLE_DECOMPOSITION)
                .components(GregoriusDrugworksMaterials.Alumina, 1, GregoriusDrugworksMaterials.SiliconDioxide, 2)
                .build();

        GregoriusDrugworksMaterials.HighTemperatureInsulation = new Material.Builder(32001, resLoc("high_temperature_insulation"))
                .dust()
                .color(0xC8C1B8)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(GregoriusDrugworksMaterials.CeramicFiber, 1, GregoriusDrugworksMaterials.SiliconDioxide, 1)
                .build();

        GregoriusDrugworksMaterials.Hexafluoropropylene = new Material.Builder(32002, resLoc("hexafluoropropylene"))
                .gas()
                .color(0xCFEFFF)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 3, Materials.Fluorine, 6)
                .build();

        GregoriusDrugworksMaterials.PerfluoroalkoxyPolymerSlurry = new Material.Builder(32003, resLoc("perfluoroalkoxy_polymer_slurry"))
                .liquid()
                .color(0xE6F2FF)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 5, Materials.Fluorine, 12, Materials.Oxygen, 1)
                .build();

        GregoriusDrugworksMaterials.FluoropolymerStructuralFilm = new Material.Builder(32004, resLoc("fluoropolymer_structural_film"))
                .polymer()
                .color(0xF7FBFF)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 5, Materials.Fluorine, 12, Materials.Oxygen, 1)
                .build();

        GregoriusDrugworksMaterials.FluoropolymerResidue = new Material.Builder(32005, resLoc("fluoropolymer_residue"))
                .liquid()
                .color(0xB6C6D6)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 3, Materials.Fluorine, 6)
                .build();

        GregoriusDrugworksMaterials.PhthalicAnhydride = new Material.Builder(32006, resLoc("phthalic_anhydride"))
                .dust()
                .color(0xF1E6D6)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 8, Materials.Hydrogen, 4, Materials.Oxygen, 3)
                .build();

        GregoriusDrugworksMaterials.TrimelliticAnhydride = new Material.Builder(32007, resLoc("trimellitic_anhydride"))
                .dust()
                .color(0xE8D7C6)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 9, Materials.Hydrogen, 4, Materials.Oxygen, 5)
                .build();

        GregoriusDrugworksMaterials.PyromelliticDianhydride = new Material.Builder(32008, resLoc("pyromellitic_dianhydride"))
                .dust()
                .color(0xE6D2B8)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 10, Materials.Hydrogen, 2, Materials.Oxygen, 6)
                .build();

        GregoriusDrugworksMaterials.Oxydianiline = new Material.Builder(32009, resLoc("oxydianiline"))
                .dust()
                .color(0xE9DAB8)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 12, Materials.Hydrogen, 12, Materials.Nitrogen, 2, Materials.Oxygen, 1)
                .build();

        GregoriusDrugworksMaterials.MPhenylenediamine = new Material.Builder(32010, resLoc("m_phenylenediamine"))
                .dust()
                .color(0xD8C7A0)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 6, Materials.Hydrogen, 8, Materials.Nitrogen, 2)
                .build();

        GregoriusDrugworksMaterials.BisphenolADianhydride = new Material.Builder(32011, resLoc("bisphenol_a_dianhydride"))
                .dust()
                .color(0xD7C1A3)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 31, Materials.Hydrogen, 18, Materials.Oxygen, 8)
                .build();

        GregoriusDrugworksMaterials.Polyetherimide = new Material.Builder(32012, resLoc("polyetherimide"))
                .liquid()
                .color(0xC79B43)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 37, Materials.Hydrogen, 22, Materials.Nitrogen, 2, Materials.Oxygen, 6)
                .build();

        GregoriusDrugworksMaterials.PolyimideSolution = new Material.Builder(32013, resLoc("polyimide_solution"))
                .liquid()
                .color(0xA06A2F)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 22, Materials.Hydrogen, 10, Materials.Nitrogen, 2, Materials.Oxygen, 5)
                .build();

        GregoriusDrugworksMaterials.MicroporousPolyimideMembrane = new Material.Builder(32014, resLoc("microporous_polyimide_membrane"))
                .polymer()
                .color(0xC9D2DA)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 22, Materials.Hydrogen, 10, Materials.Nitrogen, 2, Materials.Oxygen, 5)
                .build();

        GregoriusDrugworksMaterials.Polysiloxane = new Material.Builder(32015, resLoc("polysiloxane"))
                .liquid()
                .color(0xDDE7EF)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 2, Materials.Hydrogen, 6, Materials.Oxygen, 1, Materials.Silicon, 1)
                .build();

        GregoriusDrugworksMaterials.IronPentacarbonyl = new Material.Builder(32016, resLoc("iron_pentacarbonyl"))
                .liquid()
                .color(0xC9D4DC)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Iron, 1, Materials.Carbon, 5, Materials.Oxygen, 5)
                .build();

        GregoriusDrugworksMaterials.AsGrownMultiwalledCarbonNanotubes = new Material.Builder(32017, resLoc("as_grown_multiwalled_carbon_nanotubes"))
                .dust()
                .color(0x141414)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 1)
                .build();

        GregoriusDrugworksMaterials.PurifiedMultiwalledCarbonNanotubes = new Material.Builder(32018, resLoc("purified_multiwalled_carbon_nanotubes"))
                .dust()
                .color(0x101010)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 1)
                .build();

        GregoriusDrugworksMaterials.MDinitrobenzene = new Material.Builder(32019, resLoc("m_dinitrobenzene"))
                .dust()
                .color(0xD9C86A)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 6, Materials.Hydrogen, 4, Materials.Nitrogen, 2, Materials.Oxygen, 4)
                .build();

        GregoriusDrugworksMaterials.PChloronitrobenzene = new Material.Builder(32020, resLoc("p_chloronitrobenzene"))
                .dust()
                .color(0xD8C05E)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 6, Materials.Hydrogen, 4, Materials.Chlorine, 1, Materials.Nitrogen, 1, Materials.Oxygen, 2)
                .build();

        GregoriusDrugworksMaterials.PNitrophenol = new Material.Builder(32021, resLoc("p_nitrophenol"))
                .dust()
                .color(0xC9B04F)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 6, Materials.Hydrogen, 5, Materials.Nitrogen, 1, Materials.Oxygen, 3)
                .build();

        GregoriusDrugworksMaterials.DinitrodiphenylEther44 = new Material.Builder(32022, resLoc("dinitrodiphenyl_ether_4_4"))
                .dust()
                .color(0xC9B36B)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 12, Materials.Hydrogen, 8, Materials.Nitrogen, 2, Materials.Oxygen, 5)
                .build();

        GregoriusDrugworksMaterials.PhenolicResin = new Material.Builder(32023, resLoc("phenolic_resin"))
                .polymer()
                .color(0x6B3C22)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 7, Materials.Hydrogen, 6, Materials.Oxygen, 1)
                .build();

        GregoriusDrugworksMaterials.CarbonMatrixComposite = new Material.Builder(32024, resLoc("carbon_matrix_composite"))
                .dust()
                .color(0x1C1C1C)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Carbon, 1, Materials.Graphite, 1)
                .build();

        GregoriusDrugworksMaterials.MoltenVolcanicGlass = new Material.Builder(32025, resLoc("molten_volcanic_glass"))
                .liquid()
                .color(0x2A202B)
                .flags(DISABLE_DECOMPOSITION)
                .components(GregoriusDrugworksMaterials.SiliconDioxide, 2, GregoriusDrugworksMaterials.Alumina, 1)
                .build();

        GregoriusDrugworksMaterials.ObsidianCeramicComposite = new Material.Builder(32026, resLoc("obsidian_ceramic_composite"))
                .dust()
                .color(0x3B303E)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(GregoriusDrugworksMaterials.SiliconDioxide, 2, GregoriusDrugworksMaterials.Alumina, 1)
                .build();

        GregoriusDrugworksMaterials.AluminosilicateGel = new Material.Builder(32027, resLoc("aluminosilicate_gel"))
                .dust()
                .color(0xC8D0C8)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Aluminium, 1, Materials.Silicon, 1, Materials.Oxygen, 4, Materials.Hydrogen, 2)
                .build();

        GregoriusDrugworksMaterials.ZeoliteCatalystMatrix = new Material.Builder(32028, resLoc("zeolite_catalyst_matrix"))
                .dust()
                .color(0xB9C7B7)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(Materials.Sodium, 1, Materials.Aluminium, 1, Materials.Silicon, 1, Materials.Oxygen, 4, Materials.Hydrogen, 2)
                .build();

        GregoriusDrugworksMaterials.CatalyticCrackingMatrix = new Material.Builder(32029, resLoc("catalytic_cracking_matrix"))
                .dust()
                .color(0x879A86)
                .iconSet(DULL)
                .flags(DISABLE_DECOMPOSITION)
                .components(GregoriusDrugworksMaterials.ZeoliteCatalystMatrix, 1, GregoriusDrugworksMaterials.VanadiumPentoxide, 1, GregoriusDrugworksMaterials.Alumina, 1)
                .build();
    }
}
